### PR TITLE
TextMate grammar for *.dtd files

### LINKED ISSUE #313: TextMate grammar for *.dtd files
I think it could be helpful to provide a TextMate grammar for `*.dtd` files. Syntax highlighting would make it a bit easier to read and edit these files.

It looks like nothing is being done to implement it upstream in VSCode: [VSCode related issue](https://github.com/microsoft/vscode/issues/55751). VSCode just reuses the Atom TextMate grammar for XML/DTD: [Atom issue](https://github.com/atom/language-xml/issues/89). The Atom TextMate grammar was extracted from the [TextMate XML plugin](https://github.com/textmate/xml.tmbundle). This grammar doesn't include syntax highlighting for DTD.

### PR DESCRIPTION
**Pull request by angelozerr**
TextMate grammar for *.dtd files

Fixes #313

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,6 +97,27 @@
     "yauzl": "^2.10.0"
   },
   "contributes": {
+    "languages": [
+      {
+        "id": "dtd",
```
Comment (AlexXuChen) [inline package.json:102]: We also need to add the "onLanguage:dtd" to "activationEvents" since if we open a `.dtd` file first, XML Support won't start.

**Review by AlexXuChen — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (angelozerr) [reply package.json:103]: Good catch! Fixed.

**Review by AlexXuChen — APPROVED** [review]
LGTM

**Comment by angelozerr** [comment]
I merged the PR it is a good start, but it requires a lot of improvement. We will improve it step by step and if we have good result, we could try to contribute to vscode.

**Pull request merged**